templates/github/.github/workflows/ci.yml.j2

```diff
@@ -146,3 +146,89 @@ jobs:
           docker exec pulp ls -latr /etc/yum.repos.d/
           docker exec pulp cat /etc/yum.repos.d/*
           {%- endif %}
+
+{%- if upgrade_range %}
+
+  upgrade:
+    runs-on: ubuntu-latest
+    needs: lint
+
+    {%- if upgrade_range is iterable and upgrade_range|length > 0 %}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        {%- for test in upgrade_range %}
+          - FROM_PULPCORE_BRANCH: "{{ test.pulpcore_branch }}"
+          {%- for plugin_branch, version in test.items() if plugin_branch != "pulpcore_branch" %}
+            FROM_{{ plugin_branch| upper }}: "{{version}}"
+          {%- endfor %}
+        {%- endfor %}
+    {%- endif %}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # by default, it uses a depth of 1
+          # this fetches all history so that we can read each commit
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+
+      - name: Install httpie
+        run: |
+          echo ::group::HTTPIE
+          sudo apt-get update -yq
+          sudo -E apt-get -yq --no-install-suggests --no-install-recommends install httpie
+          echo ::endgroup::
+          echo "TEST=upgrade" >> $GITHUB_ENV
+          echo "HTTPIE_CONFIG_DIR=$GITHUB_WORKSPACE/.ci/assets/httpie/" >> $GITHUB_ENV
+          {%- for plugin_branch, version in upgrade_range[0].items() %}
```

Comment:
Why is the index 0 needed here?

Reply:
`upgrade_range` is a list of dicts:
```python
upgrade_range = [
   { "pulpcore_branch": "X.Y", "pulp_file_branch": "X.Y"},
   { "pulpcore_branch": "X.Y", "pulp_file_branch": "X.Y"}
]
```
the keys would repeat, so I took only the first dict keys